src/Importer.tsx

```diff
@@ -35,33 +35,40 @@ function Importer({ show, onImport, onClose }: ImportProps): JSX.Element {
   async function parseFile(file: File) {
     setLoading(true);
 
+    let blobs: Array<Blob> = [];
+    let games: Array<Game> = [];
+
     let blob: Blob = file;
     if (file.name.endsWith(".zip")) {
       const reader = new zip.ZipReader(new zip.BlobReader(blob));
       const entries = await reader.getEntries();
-      if (entries.length !== 1) {
-        return setError("Zip files should contain only one log file.");
+      for (let entry of entries) {
+        blobs.push(await entry.getData!(new zip.BlobWriter()));
       }
-      blob = await entries[0].getData!(new zip.BlobWriter());
+    } else {
+      blobs = [file];
     }
 
-    const parser = newParser();
-    // The type cast is required because @types/node messes with the DOM type
-    // See https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/58079
-    const stream = blob.stream() as any as ReadableStream<Uint8Array>;
-    const reader = stream.getReader();
-    const utf8Decoder = new TextDecoder("utf-8");
-    while (true) {
-      let { value, done } = await reader.read();
-      const chunk = utf8Decoder.decode(value, { stream: !done });
-      parser.parse(chunk);
-      if (done) break;
+    for (let blob of blobs) {
+      const parser = newParser();
+      // The type cast is required because @types/node messes with the DOM type
+      // See https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/58079
+      const stream = blob.stream() as any as ReadableStream<Uint8Array>;
+      const reader = stream.getReader();
+      const utf8Decoder = new TextDecoder("utf-8");
+      while (true) {
+        let { value, done } = await reader.read();
+        const chunk = utf8Decoder.decode(value, { stream: !done });
+        parser.parse(chunk);
+        if (done) break;
+      }
+      parser.end();
+
+      games.push(...parser.games());
```

Comment:
The parser operates on lines anyway, so you can just use one parser and feed it all of the zip contents in a row